pennylane/operation.py

```diff
@@ -726,6 +749,10 @@ def __str__(self):
         """Print the tensor product and some information."""
         return 'Tensor product {}: {} params, wires {}'.format([i.name for i in self.obs], len(self.params), self.wires)
 
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return "Tensor(" +", ".join([repr(o) for o in self.obs]) +")"
```

Comment:
Do we want tensors to print as `Tensor(...)` though? The actual tensor class is quite hidden from users, and we only document using `@` to create tensors. This is partly because instantiating the tensor opens us up to some known buggy edge cases, that are not possible using `@` (such as tensors of single observables)

Reply:
Hmm. I'd say do whatever serves the purpose best. `__repr__` often returns something that looks like a constructor call, but it's not a hard rule.

Reply:
Shall I change it?

Reply:
I'm on the fence, but lean towards the output matching how the user would construct it, so `@` :slightly_smiling_face: But it's not too much of an issue either way.

Reply:
We could also define both `__repr__` and `__str__`, where `__repr__` is the constructor-like looking variant and `__str__` the user-friendly one with `@`

Reply:
`__str__` is already defined, it's on line 747 just above, and it's even wordier.